db/memory/src/main/java/org/commonjava/indy/mem/data/MemoryStoreDataManager.java

```diff
@@ -205,12 +208,12 @@ private void recurseGroup( final Group master, final List<ArtifactStore> result,
                 if ( recurseGroups && type == StoreType.group )
                 {
                     // if we're here, we're definitely recursing groups...
-                    recurseGroup( (Group) stores.get( key ), result, seen, includeGroups, true );
+                    recurseGroup( (Group) stores.get( key ), result, seen, includeGroups, true, enabledOnly );
                 }
                 else
                 {
                     final ArtifactStore store = stores.get( key );
-                    if ( store != null )
+                    if ( store != null && !( store.isDisabled() && enabledOnly ) )
                     {
```

Comment:
Can we add a few unit tests for this logic, just for MemoryStoreDataManager (since everything is based on that)? Some of this logic is just hard enough to parse that I think it could become a source of regressions if we don't have it covered.